composeApp/src/main/kotlin/io/github/couchtracker/utils/Loadable.kt

```diff
@@ -1,39 +1,71 @@
 package io.github.couchtracker.utils
 
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
 import androidx.compose.runtime.State
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
 import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.ExperimentalCoroutinesApi
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.map
 
 /**
  * Extension of [Result] that allows the value to be [Loading].
  */
-sealed interface Loadable<out T, out E> : Actionable<T, E> {
+sealed interface Loadable<out T> : Actionable<T, Nothing> {
 
-    data object Loading : Loadable<Nothing, Nothing>
+    data object Loading : Loadable<Nothing>
+    data class Loaded<T>(val value: T) : Loadable<T>
 }
 
 /**
- * Applies [f] to [Result.Value.value].
- * Otherwise, [Loadable.Loading] and [Result.Error] are returned without executing [f].
+ * Applies [f] to [Loadable.Loaded.value].
+ * Otherwise, [Loadable.Loading] are returned without executing [f].
  */
-inline fun <I, O, E> Loadable<I, E>.flatMap(f: (I) -> Loadable<O, E>): Loadable<O, E> = when (this) {
+inline fun <I, O> Loadable<I>.flatMap(f: (I) -> Loadable<O>): Loadable<O> = when (this) {
     is Loadable.Loading -> this
-    is Result.Error -> this
-    is Result.Value -> f(value)
+    is Loadable.Loaded -> f(value)
 }
 
 /**
- * Applies [f] to [Result.Value.value].
- * Otherwise, [Loadable.Loading] and [Result.Error] are returned without executing [f].
+ * Applies [f] to [Loadable.Loaded.value].
+ * Otherwise, [Loadable.Loading] are returned without executing [f].
  */
-inline fun <I, O, E> Loadable<I, E>.map(f: (I) -> O): Loadable<O, E> = flatMap {
-    Result.Value(f(it))
+inline fun <I, O> Loadable<I>.map(f: (I) -> O): Loadable<O> = flatMap {
+    Loadable.Loaded(f(it))
 }
 
+/**
+ * Returns [Loadable.Loaded.value], or `null` in other cases.
+ */
+val <T> Loadable<T>.valueOrNull: T?
```

Comment:
I think it's better if these are if these are extension functions instead of extension properties, to be more aligned with other stdlib functions.